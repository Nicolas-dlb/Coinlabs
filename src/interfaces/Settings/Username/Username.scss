.settings_username {
  margin-bottom: 2vh;
  width: 85%;
  align-self: flex-end;
  display: flex;
  flex-direction: column;
  align-items: flex-start;
  justify-content: flex-start;
  padding: 20px;
  border-radius: 15px;
  background: var(--third-bg-color);
  transition: background-color 0.3s ease;
  margin-top: 2vh;
  .intro_change {
    margin-top: 2vh;
    color: var(--third-font-color);
  }
  h3 {
    margin: 0vh 0;
  }
  .settings_username_input {
    display: flex;
    align-items: center;
    width: 100%;
    button {
      margin-top: 13px;
      margin-left: 15px;
    }
  }
  .input {
    display: flex;
    margin-top: 2vh;
    border: 1px solid var(--third-font-color);
    border-radius: 10px;
    height: 35px;
    padding-left: 10px;
    width: 59%;
    align-items: center;
    p {
      font-size: 18px;
    }
    input {
      background: var(--third-bg-color);
      border: none;
      outline: none;
      transition: background-color 0.3s ease;
      color: var(--primary-font-color);
      caret-color: #1baa63;
      border-radius: 10px;
      font-family: inherit;
      padding: 8px;
      width: 100px;
      font-size: 16px;
      &::placeholder {
        font-family: inherit;
        color: #484558;
        font-size: 12px;
      }
    }
  }
}
