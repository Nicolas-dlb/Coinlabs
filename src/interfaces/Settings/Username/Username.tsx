import React from "react";
import "./Username.scss";

type UsernameProps = {
  onClick: () => void;
};

function Username({ onClick }: UsernameProps) {
  return (
    <div className="settings_username">
      <h3>Username</h3>

      <div className="settings_username_input">
        <div className="input">
          <p>@</p>
          <input
            placeholder="Enter username"
            type="text"
            id="change_username"
            className="username_element"
            spellCheck="false"
            autoComplete="off"
            onKeyPress={(e) => {
              if (e.key === "Enter") {
                onClick();
              }
            }}
          />
        </div>
        <button className="btn_settings" type="button" onClick={onClick}>
          Update
        </button>
      </div>
    </div>
  );
}

export default Username;
