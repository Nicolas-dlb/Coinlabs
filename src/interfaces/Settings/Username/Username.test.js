import React from "react";
import { render } from "enzyme";
import { act } from "react-dom/test-utils";
import Username from "./Username";

describe("Username", () => {
  let UsernameComponent;
  beforeEach(async () => {
    await act(async () => {
      UsernameComponent = render(<Username />);
    });
  });
  it("should render correctly", () => {
    expect(UsernameComponent).toMatchSnapshot();
  });
  it("should render Username title", () => {
    expect(UsernameComponent.find("h3").text()).toBe("Username");
  });
  it("should render an icon", () => {
    expect(UsernameComponent.find("p").text()).toBe("@");
  });

  it("should render an input", () => {
    expect(UsernameComponent.find("input").length).toBe(1);
  });
  it("should render a button called Update", () => {
    expect(UsernameComponent.find("button").text()).toBe("Update");
  });
});
