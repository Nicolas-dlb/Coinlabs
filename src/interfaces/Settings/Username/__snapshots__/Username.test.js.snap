// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Username should render correctly 1`] = `
<div
  class="settings_username"
>
  <h3>
    Username
  </h3>
  <div
    class="settings_username_input"
  >
    <div
      class="input"
    >
      <p>
        @
      </p>
      <input
        autocomplete="off"
        class="username_element"
        id="change_username"
        placeholder="Enter username"
        spellcheck="false"
        type="text"
      />
    </div>
    <button
      class="btn_settings"
      type="button"
    >
      Update
    </button>
  </div>
</div>
`;
